### DISCUSSION TITLE
Execute context once on startup / CloudFlare Workers

### DISCUSSION THREAD

**Discussion by sntmt** [honojs/hono#1030] [Q&A] [answered]
Hello,

I'm seeing that for get the environment variables as bindings, you use the context from the app[method], but I would like some like dotenv, where you can get the variable calling directly to the environment.

I have this, a middleware that get the context and on every request, get the variables from the wrangler.toml and is saved in a static class, so I can get the variables:

```
// index.ts
app.use('*', setEnvironmentVariables())
```

```
// setEnvironmentVariables.ts
import { MiddlewareHandler } from 'hono'
import { Vars } from '../utils/constants'

export const setEnvironmentVariables = (): MiddlewareHandler => {
	return async (ctx, next) => {
		Vars.set(ctx,
			["VAR1", "VAR2"])
		await next()
	}
}
```
```
// constants.ts

import { Context } from 'hono'
export class Vars {
	private static constants = new Map()
	static set(c: Context, vars: string[]): void {
		
		vars.forEach((key) => {
			if (Object.keys(c.env).includes(key)) {
				this.constants.set(key, c.env[key] || '')
			} else {
				console.log(`error getting environment variable ${key}`)
			}
		})
	}
	static get(key: string): string {
		return this.constants.get(key)
	}
}
````
```
// wrangler.toml
[vars]
VAR1 = VALUE1
VAR2 = VALUE2
````
So in the code where I want to use the variable:

```
Vars.get('VAR1') // VALUE1
```

Question is, can I execute the code inside the middleware as function to execute once on the startup of the worker? because being executed only once to set the variables would be enough.

Thank you

**Comment by yusukebe** [comment, marked as answer]
Hi @sntmt !

You can't do it, and you don't have to do it if you want the application to run on Cloudflare Workers. On Cloudflare Workers, a Worker is launched for each request. It is not guaranteed that a Worker is always running. So, even if it has a value as static, it may disappear, so it must be retrieved from Bindings each time.

**Reply by sntmt** [reply]
Thanks @yusukebe!

**Comment by muhaimincs** [comment]
Hi. I just implement this library today and I found that it's hard to get the right documentation to read the `.env` file. Where do I start?

**Reply by yusukebe** [reply]
https://developers.cloudflare.com/workers/configuration/environment-variables/
https://hono.dev/getting-started/cloudflare-workers#bindings
